uwblueprint/ccbc
# Alex/publish review function

## Notion ticket link

[Publish Review Function](https://www.notion.so/uwblueprintexecs/Publish-Review-Function-2f0d92e227c84cf2922c4f4b0e82bcb2)



## Implementation description
* Publish review button on Create Review page brings up a modal that, when clicked, will publish a user's review
* React Context for notifications added, a published toast will appear on the dashboard page when the review is successfully published, an error toast will appear on the create review page if an error occurs



## Steps to test
1. The component can only be tested with the dummy book since the modal to add a book is missing
2. Navigate to the `/create-review` route or click on "Add Review" on the dashboard
3. Enter review details (review, byline) and click on the publish button
4. The page should redirect to the Admin Dashboard page with a blue info toast in the bottom right corner

<img width="1792" alt="Screen Shot 2022-03-06 at 7 02 04 PM" src="https://user-images.githubusercontent.com/33358619/156948163-b0645fee-67ad-4ed1-9d49-cde7d46eea4f.png">

5. Note: publishing a review with 0 books is a valid request, however, this seems to cause an error on the Admin Dashboard page


## Checklist
